src/Parser/Shaman/Elemental/Modules/Talents/Stormkeeper.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+
+const affectedAbilities = [SPELLS.LIGHTNING_BOLT_OVERLOAD.id,
+                           SPELLS.LIGHTNING_BOLT.id,
+                           SPELLS.CHAIN_LIGHTNING_OVERLOAD.id,
+                           SPELLS.CHAIN_LIGHTNING.id];
+
+class Stormkeeper extends Analyzer {
+  damageGained = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.STORMKEEPER_TALENT.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if (!this.selectedCombatant.hasBuff(SPELLS.STORMKEEPER_TALENT.id)){
+      return;
+    }
+
+    if (!affectedAbilities.includes(event.ability.guid)) {
+      return;
+    }
+    this.damageGained+=event.amount;
+  }
+
+  get damagePercent() {
+    return this.owner.getPercentageOfTotalDamageDone(this.damageGained);
+  }
+
+  get damagePerSecond() {
+    return this.damageGained / (this.owner.fightDuration / 1000);
+  }
+
+  statistic() {
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.STORMKEEPER_TALENT.id} />}
+        value={`~ ${formatPercentage(this.damagePercent)} %`}
+        label="Of total damage"
```

Comment:
I think this estimate is way higher than the actual damage contributed by the talent. The cast time of Stormkeeper + instant cast 2 CL/LB is 3 GCDs or 4.5s before haste. In that time you could've cast 2.25 CL/LB. So you lose 0.25 CL/LB worth of damage, and gain the extra elemental overload. I'm not sure how the overload from Stormkeeper works, but I'm guessing it is just makes overload 100% chance to proc, so you'd also have to subtract the base chance to overload to get a better estimate.

Reply:
I am not completely sure which would be preferable in that case.
I was thinking that in b that case knowing how much damage the "special casts" were doing would be preferable to doing an estimation on how much damage you might have done in the same time and calculating a damage gain

Reply:
You can do that if you want, it just has to be very clear from the display that is what is shown. Usually anything showing % dmg on the site aims to show the percentage gain from picking a talent/trait/etc.